pkg/minishift/setup/hypervisor/virtualization_darwin.go

```diff
@@ -17,15 +17,80 @@ limitations under the License.
 package hypervisor
 
 import (
+	"errors"
 	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+
+	"github.com/minishift/minishift/pkg/util/progressdots"
+)
+
+const (
+	driverBinaryDir   = "/usr/local/bin"
+	driverBinaryPath  = driverBinaryDir + "/docker-machine-driver-xhyve"
+	driverDownloadUrl = "https://github.com/machine-drivers/docker-machine-driver-xhyve/releases/download/v0.3.3/docker-machine-driver-xhyve"
 )
 
 func CheckHypervisorAvailable() error {
-	fmt.Println("Checking Hypervisor Availability ...")
 	return nil
 }
 
 func CheckAndConfigureHypervisor() error {
-	fmt.Println("Checking and Configuring Hypervisor  ...")
+	if isRoot() {
+		fmt.Print("Configuring Xhyve Hypervisor.")
+		progressDots := progressdots.New()
+		progressDots.Start()
+		err := downloadXhyveDriver(driverBinaryPath, driverDownloadUrl)
+		progressDots.Stop()
+		fmt.Print("OK")
+		return err
+	}
+	return errors.New("You need to execute this command with sudo.")
+}
+
+func downloadXhyveDriver(filepath string, url string) error {
+	os.MkdirAll(driverBinaryDir, 0751)
+
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	err = out.Chown(0, 0)
+	if err != nil {
+		return err
+	}
+
+	err = out.Chmod(os.ModeSetuid | 0751)
+	if err != nil {
+		return err
+	}
+
```

Comment:
```
$ sudo minishift setup 
-- minishift version: v1.28.0+7bbed531
Configuring Xhyve Hypervisor.....OK

Consider checking getting started guide at https://docs.okd.io/latest/minishift/getting-started/index.html
```

The above output is not telling about what are the steps are being done in the background. We should  be transparent about changes we are doing to the system if user needs to know them. Because if user wants to cleanup the installation then at-least he would be aware of what changes were done to the system for Minishift setup.  Also that would make it consistent with `minishift setup` output in Windows.

Reply:
Also when I run it when `the docker-machine-driver-xhyve` already present with correctly filepermission we should say so.e.g. `docker-machine-driver-xhyve` exists in the PATH with right permissions`.

Reply:
@LalatenduMohanty I deliberately did this because, we have the preflight check which already checks for this, my understanding is that, the user will only run the `minishift setup` command if `minishift start` fails for him in the first place, so checking again if the requirements are met felt like an redundant task. So `minishift setup` will setup the system again even if the requirements are already met.

Reply:
> my understanding is that, the user will only run the `minishift setup` command if `minishift start` fails for him in the first place

I do not agree. Once `minishift setup` is ready we will update our documentation to suggest running `minishift setup` first then `minishift start`. Same applicable when vscode openshift plugin suggest users to install minishift or even automates the install part.